### ISSUE TITLE
Upgrade from 0.8.0 to 0.8.2 but meet insufficient funds then stop

### ISSUE THREAD

**Issue by lcgogo** [crypto-org-chain/cronos#751]
I stop 0.8.0 node, add followed config then start 0.8.2 node
```
# IAVLDisableFastNode enables or disables the fast node feature of IAVL.
# Default is false.
iavl-disable-fastnode = true
```

But meet followed error and the node stops.
```shell
1:32PM ERR Stopping peer for error err=EOF module=p2p peer={"Data":{},"Logger":{}} server=node
1:32PM INF service stop impl={"Data":{},"Logger":{}} module=p2p msg={} peer={"id":"4c37797ccc4f84fc5a910d6d29ea874282c70792","ip":"46.137.250.193","port":26656} server=node
1:32PM INF service start impl="Peer{MConn{18.141.34.83:26656} 04c28545b8fca79eebd9081f731cec92c0a6536f out}" module=p2p msg={} peer={"id":"04c28545b8fca79eebd9081f731cec92c0a6536f","ip":"18.141.34.83","port":26656} server=node
1:32PM INF service start impl=MConn{18.141.34.83:26656} module=p2p msg={} peer={"id":"04c28545b8fca79eebd9081f731cec92c0a6536f","ip":"18.141.34.83","port":26656} server=node
1:32PM INF Added peer module=p2p peer={"Data":{},"Logger":{}} server=node
1:32PM INF Connection is closed @ recvRoutine (likely by the other side) conn={"Logger":{}} module=p2p peer={"id":"04c28545b8fca79eebd9081f731cec92c0a6536f","ip":"18.141.34.83","port":26656} server=node
1:32PM INF service stop impl={"Logger":{}} module=p2p msg={} peer={"id":"04c28545b8fca79eebd9081f731cec92c0a6536f","ip":"18.141.34.83","port":26656} server=node
1:32PM ERR Stopping peer for error err=EOF module=p2p peer={"Data":{},"Logger":{}} server=node
1:32PM INF service stop impl={"Data":{},"Logger":{}} module=p2p msg={} peer={"id":"04c28545b8fca79eebd9081f731cec92c0a6536f","ip":"18.141.34.83","port":26656} server=node
panic: 13110810582431757052basecro is smaller than 16186523217552194547basecro: insufficient funds

goroutine 92 [running]:
github.com/cosmos/cosmos-sdk/x/distribution/keeper.Keeper.AllocateTokens({{0x2e03c98, 0xc001124b40}, {0x2e24ca0, 0xc00133c280}, {{0x2e24ca0, 0xc00133c280}, 0xc00011c390, {0x2e03c98, 0xc001124b80}, {0x2e03ce8, ...}, ...}, ...}, ...)
	/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.9/x/distribution/keeper/allocation.go:32 +0x1450
github.com/cosmos/cosmos-sdk/x/distribution.BeginBlocker({{0x2e18618, 0xc0001a8000}, {0x2e26880, 0xc003c82000}, {{0xb, 0x0}, {0xc005cc3110, 0x12}, 0x50b14d, {0x81f8317, ...}, ...}, ...}, ...)
	/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.9/x/distribution/abci.go:32 +0x211
github.com/cosmos/cosmos-sdk/x/distribution.AppModule.BeginBlock(...)
	/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.9/x/distribution/module.go:169
github.com/cosmos/cosmos-sdk/types/module.(*Manager).BeginBlock(_, {{0x2e18618, 0xc0001a8000}, {0x2e26880, 0xc003c82000}, {{0xb, 0x0}, {0xc005cc3110, 0x12}, 0x50b14d, ...}, ...}, ...)
	/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.9/types/module/module.go:491 +0x3bb
github.com/crypto-org-chain/cronos/app.(*App).BeginBlocker(_, {{0x2e18618, 0xc0001a8000}, {0x2e26880, 0xc003c82000}, {{0xb, 0x0}, {0xc005cc3110, 0x12}, 0x50b14d, ...}, ...}, ...)
	/cro-src/cronos-0.8.2/app/app.go:735 +0x85
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).BeginBlock(_, {{0xc003acb800, 0x20, 0x20}, {{0xb, 0x0}, {0xc005cc3110, 0x12}, 0x50b14d, {0x81f8317, ...}, ...}, ...})
	/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.45.9/baseapp/abci.go:177 +0x97c
github.com/tendermint/tendermint/abci/client.(*localClient).BeginBlockSync(_, {{0xc003acb800, 0x20, 0x20}, {{0xb, 0x0}, {0xc005cc3110, 0x12}, 0x50b14d, {0x81f8317, ...}, ...}, ...})
	/go/pkg/mod/github.com/tendermint/tendermint@v0.34.20/abci/client/local_client.go:280 +0x118
github.com/tendermint/tendermint/proxy.(*appConnConsensus).BeginBlockSync(_, {{0xc003acb800, 0x20, 0x20}, {{0xb, 0x0}, {0xc005cc3110, 0x12}, 0x50b14d, {0x81f8317, ...}, ...}, ...})
	/go/pkg/mod/github.com/tendermint/tendermint@v0.34.20/proxy/app_conn.go:81 +0x55
github.com/tendermint/tendermint/state.execBlockOnProxyApp({0x2e195a0?, 0xc00197c1e0}, {0x2e1f6f0, 0xc0005a3010}, 0xc0037a3a40, {0x2e256b0, 0xc0012b3f00}, 0x50b14c?)
	/go/pkg/mod/github.com/tendermint/tendermint@v0.34.20/state/execution.go:307 +0x3dd
github.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(_, {{{0xb, 0x0}, {0xc006ed9704, 0x7}}, {0xc002baaf60, 0x12}, 0x1, 0x50b14c, {{0xc0035a2380, ...}, ...}, ...}, ...)
	/go/pkg/mod/github.com/tendermint/tendermint@v0.34.20/state/execution.go:140 +0x171
github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).poolRoutine(0xc000f48700, 0x0)
	/go/pkg/mod/github.com/tendermint/tendermint@v0.34.20/blockchain/v0/reactor.go:404 +0xbda
created by github.com/tendermint/tendermint/blockchain/v0.(*BlockchainReactor).OnStart
	/go/pkg/mod/github.com/tendermint/tendermint@v0.34.20/blockchain/v0/reactor.go:110 +0x7a
```

**Comment by yihuang** [comment]
Looks like binary with wrong network.

**Comment by mmsqe** [comment]
@lcgogo Do you remember the height when restart?

**Comment by yihuang** [comment]
> v0.8.3 is a non-breaking upgrade that fixes an app hash mismatch issue caused by switching iavl-disable-fastnode option.

Should have fixed in [`v0.8.3`](https://github.com/crypto-org-chain/cronos/releases/tag/v0.8.3)

**Comment by lcgogo** [comment]
> > v0.8.3 is a non-breaking upgrade that fixes an app hash mismatch issue caused by switching iavl-disable-fastnode option.
> 
> Should have fixed in [`v0.8.3`](https://github.com/crypto-org-chain/cronos/releases/tag/v0.8.3)

Thanks, tried 0.8.3 now it's ok.
